### PR TITLE
Fix gatk/markduplicatesspark

### PR DESCRIPTION
**Pull request by SusiJo**
## PR checklist

This PR fixes the issue (see below) that arises from `docker.userEmulation = true` and can be solved by overriding that setting for this very module. Additionally, the testdata reference genomes and bam files were mixed up (chr21 and chr22) and led to failing tests. My suggestion would be to add the corresponding chromosome string to the name of the files to avoid confusion. 
The current testdata that is used, leads to another bug, but the module is at least starting to run now. 
`A USER ERROR has occurred: Multiple inputs to MarkDuplicatesSpark detected. MarkDuplicatesSpark requires all inputs to be queryname sorted or querygroup-sorted for multi-input processing but input test.paired_end.sorted.bam was sorted in SAMFileHeader{VN=1.6, SO=coordinate} order`

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

The `userEmulation = true` leads to this Error:
```
Command error:
  A USER ERROR has occurred: Failed to read bam header from test.paired_end.sorted.bam
   Caused by:failure to login: javax.security.auth.login.LoginException: java.lang.NullPointerException: invalid null input: name
        at com.sun.security.auth.UnixPrincipal.<init>(UnixPrincipal.java:71)
        at com.sun.security.auth.module.UnixLoginModule.login(UnixLoginModule.java:133)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at javax.security.auth.login.LoginContext.invoke(LoginContext.java:755)
        at javax.security.auth.login.LoginContext.access$000(LoginContext.java:195)
        at javax.security.auth.login.LoginContext$4.run(LoginContext.java:682)
        at javax.security.auth.login.LoginContext$4.run(LoginContext.java:680)
        at java.security.AccessController.doPrivileged(Native Method)
        at javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:680)
        at javax.security.auth.login.LoginContext.login(LoginContext.java:587)
        at org.apache.hadoop.security.UserGroupInformation$HadoopLoginContext.login(UserGroupInformation.java:1926)
        at org.apache.hadoop.security.UserGroupInformation.doSubjectLogin(UserGroupInformation.java:1837)
        at org.apache.hadoop.security.UserGroupInformation.createLoginUser(UserGroupInformation.java:710)
        at org.apache.hadoop.security.UserGroupInformation.getLoginUser(UserGroupInformation.java:660)
        at org.apache.hadoop.security.UserGroupInformation.getCurrentUser(UserGroupInformation.java:571)
        at org.apache.hadoop.fs.FileSystem$Cache$Key.<init>(FileSystem.java:3487)
        at org.apache.hadoop.fs.FileSystem$Cache$Key.<init>(FileSystem.java:3477)
        at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:3319)
        at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:479)
        at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:227)
        at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:463)
        at org.apache.hadoop.fs.Path.getFileSystem(Path.java:365)
        at org.disq_bio.disq.impl.file.HadoopFileSystemWrapper.isDirectory(HadoopFileSystemWrapper.java:101)
        at org.disq_bio.disq.HtsjdkReadsRddStorage.read(HtsjdkReadsRddStorage.java:148)
        at org.disq_bio.disq.HtsjdkReadsRddStorage.read(HtsjdkReadsRddStorage.java:127)
        at org.broadinstitute.hellbender.engine.spark.datasources.ReadsSparkSource.getHeader(ReadsSparkSource.java:188)
        at org.broadinstitute.hellbender.engine.spark.GATKSparkTool.initializeReads(GATKSparkTool.java:575)
        at org.broadinstitute.hellbender.engine.spark.GATKSparkTool.initializeToolInputs(GATKSparkTool.java:554)
        at org.broadinstitute.hellbender.engine.spark.GATKSparkTool.runPipeline(GATKSparkTool.java:544)
        at org.broadinstitute.hellbender.engine.spark.SparkCommandLineProgram.doWork(SparkCommandLineProgram.java:31)
        at org.broadinstitute.hellbender.cmdline.CommandLineProgram.runTool(CommandLineProgram.java:140)
        at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMainPostParseArgs(CommandLineProgram.java:192)
        at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:211)
        at org.broadinstitute.hellbender.Main.runCommandLineProgram(Main.java:160)
        at org.broadinstitute.hellbender.Main.mainEntry(Main.java:203)
        at org.broadinstitute.hellbender.Main.main(Main.java:289)
```

### CODE REVIEW

**Comment by FriederikeHanssen** [comment]
@SusiJo you are my personal hero here 🦸‍♀️ . Let's check with the modules team when and why the docker userEmualtion was added.

**Comment by SusiJo** [comment]
Apparently, there was a really long discussion about that on slack in `modules` 👀

File: modules/gatk4/markduplicatesspark/main.nf [hunk 1]
```diff
@@ -14,25 +14,26 @@ process GATK4_MARKDUPLICATES_SPARK {
...
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
-    prefix = task.ext.prefix ?: "${meta.id}"
+    def prefix = task.ext.prefix ?: "${meta.id}" + ".bam"
     def input_list = bam.collect{"--input $it"}.join(' ')
 
+
```
Comment (maxulysse) [inline modules/gatk4/markduplicatesspark/main.nf:29]:
```suggestion
```

File: tests/modules/gatk4/markduplicatesspark/nextflow.config [hunk 2]
```diff
@@ -2,4 +2,10 @@ process {
 
     publishDir = { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" }
 
+    withName: GATK4_MARKDUPLICATES_SPARK_METRICS {
+        ext.args = '--metrics-file test.metrics'
+    }
 }
+// override tests/config/nextflow.config
+docker.userEmulation = false
```
Comment (maxulysse) [inline tests/modules/gatk4/markduplicatesspark/nextflow.config:10]: no issues with singularity?

Comment (FriederikeHanssen) [reply tests/modules/gatk4/markduplicatesspark/nextflow.config:10]: nope

**Review by FriederikeHanssen — APPROVED** [review]

**Pull request merged**